src/V3MergeCond.cpp

```diff
@@ -0,0 +1,236 @@
+            return condp;
+        } else if (AstAnd* const andp = VN_CAST(rhsp, And)) {
+            if (AstNodeCond* const condp = VN_CAST(andp->rhsp(), NodeCond)) {
+                if (VN_IS(andp->lhsp(), Const)) { return condp; }
+            }
+        }
+        return NULL;
+    }
+
+    // Fold the RHS expression assuming the given condition state. Unlink bits
+    // from the RHS which is only used once, and can be reused. What remains
+    // of the RHS is expected to be deleted by the caller.
+    AstNode* foldAndUnlink(AstNode* rhsp, bool condTrue) {
+        if (rhsp->sameTree(m_mgCondp)) {
+            return condTrue ? new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 1)
+                            : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+        } else if (AstNodeCond* const condp = extractCond(rhsp)) {
+            AstNode* const resp
+                = condTrue ? condp->expr1p()->unlinkFrBack() : condp->expr2p()->unlinkFrBack();
+            if (condp == rhsp) { return resp; }
+            if (AstAnd* const andp = VN_CAST(rhsp, And)) {
+                UASSERT_OBJ(andp->rhsp() == condp, rhsp, "Should not try to fold this");
+                return new AstAnd(andp->fileline(), andp->lhsp()->cloneTree(false), resp);
+            }
+        } else if (AstAnd* const andp = VN_CAST(rhsp, And)) {
+            if (andp->lhsp()->sameTree(m_mgCondp)) {
+                return condTrue ? andp->rhsp()->unlinkFrBack()
+                                : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+            }
+            if (andp->rhsp()->sameTree(m_mgCondp)) {
+                return condTrue ? andp->lhsp()->unlinkFrBack()
+                                : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+            }
+        }
+        rhsp->v3fatal("Don't know how to fold expression");
+    }
+
+    void mergeEnd() {
+        UASSERT(m_mgFirstp, "mergeEnd without list");
+        // Merge if list is longer than one node
+        if (m_mgFirstp != m_mgLastp) {
+            UINFO(6, "MergeCond - First: " << m_mgFirstp << " Last: " << m_mgLastp << endl);
+            ++m_statMerges;
+
+            // Create equivalent 'if' statement and insert it before the first node
+            AstIf* const ifp
+                = new AstIf(m_mgCondp->fileline(), m_mgCondp->unlinkFrBack(), NULL, NULL);
+            m_mgFirstp->replaceWith(ifp);
+            ifp->addNextHere(m_mgFirstp);
+            // Unzip the list and insert under branches
+            AstNode* nextp = m_mgFirstp;
+            do {
+                // Grab next pointer and unlink
+                AstNode* const currp = nextp;
+                nextp = currp != m_mgLastp ? currp->nextp() : NULL;
+                currp->unlinkFrBack();
+                // Skip over comments
+                if (VN_IS(currp, Comment)) {
+                    currp->deleteTree();
```

Comment:
```suggestion
                    VL_DO_DANGLING(currp->deleteTree(), curp);
```

Reply:
1. I wish C++ supported "var = delete" meaning any further use would be an error.

2. Given the choice between const and DANGLING I'd probably pick dangling as finds more bugs (perhaps is personal style).

3. Try this. A little hacky but should do the job.  Maybe there's a better way, but can't just cast away const as don't know data type (without a newer C++ AFIAK):
```
diff --git a/include/verilatedos.h b/include/verilatedos.h
--- a/include/verilatedos.h
+++ b/include/verilatedos.h
@@ -153,9 +153,10 @@
 # define VL_DANGLING(var)
 #else
 ///< After e.g. delete, set variable to NULL to indicate must not use later
+// A reinterpret_cast will not work here as must also cast-away const
 # define VL_DANGLING(var) \
     do { \
-        (var) = NULL; \
+        *((void**)(&(var))) = NULL; \
     } while (false)
 #endif


```

Reply:
Let me know if that works I'll test & commit for you.

Reply:
I wish C++ was Scala.

That patch does compile, so does this if you want to maximize the occurrences of the substring const on a single line for comical effect:
```
  ///< After e.g. delete, set variable to NULL to indicate must not use later
   # define VL_DANGLING(var) \
       do { \
  -        (var) = NULL; \
  +        *const_cast<const void**>(reinterpret_cast<const void* const*>(&var)) = 0; \
       } while (false)
   #endif
```